<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>svntogit/packages.git - Git clone of the 'packages' repository
</title>
<meta name='generator' content='cgit v0.10.2'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://projects.archlinux.org/svntogit/packages.git/atom/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux' type='application/atom+xml'/>
<link rel='vcs-git' href='git://projects.archlinux.org/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='http://projects.archlinux.org/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='https://projects.archlinux.org/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='ssh://gerolde.archlinux.org/srv/projects/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
</head>
<body>
	<div id="archnavbar"><!-- Arch Linux global navigation bar -->
		<div id="archnavbarlogo">
			<p><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more"></a></p>
		</div>
		<div id="archnavbarmenu">
			<ul id="archnavbarlist">
				<li id="anb-home"><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
				<li id="anb-packages"><a href="http://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
				<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
				<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
				<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and follow bugs">Bugs</a></li>
				<li id="anb-aur"><a href="https://aur.archlinux.org/" title="Arch Linux User Repository">AUR</a></li>
				<li id="anb-download"><a href="http://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
			</ul>
		</div>
	</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'><a href='/'>index</a> : <a title='svntogit/packages.git' href='/svntogit/packages.git/'>svntogit/packages.git</a></td></tr>
<tr><td class='sub'>Git clone of the 'packages' repository
</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/svntogit/packages.git/?h=packages/linux'>summary</a><a href='/svntogit/packages.git/refs/?h=packages/linux'>refs</a><a href='/svntogit/packages.git/log/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>log</a><a class='active' href='/svntogit/packages.git/tree/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>tree</a><a href='/svntogit/packages.git/commit/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>commit</a><a href='/svntogit/packages.git/diff/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>diff</a><a href='/svntogit/packages.git/stats/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>stats</a></td><td class='form'><form class='right' method='get' action='/svntogit/packages.git/log/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch'>
<input type='hidden' name='h' value='packages/linux'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/svntogit/packages.git/tree/?h=packages/linux'>root</a>/<a href='/svntogit/packages.git/tree/trunk?h=packages/linux'>trunk</a>/<a href='/svntogit/packages.git/tree/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch</a></div><div class='content'>blob: fcad0c0b95920b8fc840853aab54590e7ca40dc1 (<a href='/svntogit/packages.git/plain/trunk/0002-block-loop-avoiding-too-many-pending-per-work-I-O.patch?h=packages/linux'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
</pre></td>
<td class='lines'><pre><code>From 63eb7031005b6f22fc1dab9f0c29435f1b52a96b Mon Sep 17 00:00:00 2001
From: Ming Lei &lt;ming.lei&#64;canonical.com&gt;
Date: Tue, 5 May 2015 19:49:55 +0800
Subject: [PATCH 2/2] block: loop: avoiding too many pending per work I/O

If there are too many pending per work I/O, too many
high priority work thread can be generated so that
system performance can be effected.

This patch limits the max_active parameter of workqueue as 16.

This patch fixes Fedora 22 live booting performance
regression when it is booted from squashfs over dm
based on loop, and looks the following reasons are
related with the problem:

<span class="hl kwb">- not like other filesyststems(such as ext4), squashfs</span>
is a bit special, and I observed that increasing I/O jobs
to access file in squashfs only improve I/O performance a
little, but it can make big difference for ext4

<span class="hl kwb">- nested loop: both squashfs.img and ext3fs.img are mounted</span>
as loop block, and ext3fs.img is inside the squashfs

<span class="hl kwb">- during booting, lots of tasks may run concurrently</span>

Fixes: b5dd2f6047ca108001328aac0e8588edd15f1778
Cc: stable&#64;vger.kernel.org (v4.0)
Cc: Justin M. Forbes &lt;jforbes&#64;fedoraproject.org&gt;
Signed-off-by: Ming Lei &lt;ming.lei&#64;canonical.com&gt;
Acked-by: Tejun Heo &lt;tj&#64;kernel.org&gt;
Signed-off-by: Jens Axboe &lt;axboe&#64;fb.com&gt;
<span class="hl kwb">---</span>
 drivers/block/loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/loop.c b/drivers/block/loop.c
index 9786aa3..cef6fa8 100644
<span class="hl kwb">--- a/drivers/block/loop.c</span>
<span class="hl kwa">+++ b/drivers/block/loop.c</span>
&#64;&#64; -725,7 +725,7 &#64;&#64; static int loop_set_fd(struct loop_device *lo, fmode_t mode,
 		goto out_putf;
 	error = -ENOMEM;
 	lo-&gt;wq = alloc_workqueue(&quot;kloopd%d&quot;,
<span class="hl kwb">-			WQ_MEM_RECLAIM | WQ_HIGHPRI | WQ_UNBOUND, 0,</span>
<span class="hl kwa">+			WQ_MEM_RECLAIM | WQ_HIGHPRI | WQ_UNBOUND, 16,</span>
 			lo-&gt;lo_number);
 	if (!lo-&gt;wq)
 		goto out_putf;
<span class="hl kwb">-- </span>
2.4.5

</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class="foot" style="padding-left:1em;padding-right:1em;">
<p>Copyright &copy; 2002-2014 <a href="mailto:jvinet@zeroflux.org"
title="contact Judd Vinet">Judd Vinet</a> and <a href="mailto:aaron@archlinux.org"
title="contact Aaron Griffin">Aaron Griffin</a>. The Arch Linux name and logo
are recognized trademarks. Some rights reserved. The registered trademark
Linux&reg; is used pursuant to a sublicense from LMI, the exclusive licensee
of Linus Torvalds, owner of the mark on a world-wide basis.</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
